sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LogLevel.java

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.logging;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Enum which represent logging levels used in Azure SDKs.
+ */
+public enum LogLevel {
+    /**
+     * Indicates that log level is at verbose level.
+     */
+    VERBOSE(1, "verbose"),
+
+    /**
+     * Indicates that log level is at information level.
+     */
+    INFORMATIONAL(2, "info", "information", "informational"),
+
+    /**
+     * Indicates that log level is at warning level.
+     */
+    WARNING(3, "warn", "warning"),
+
+    /**
+     * Indicates that log level is at error level.
+     */
+    ERROR(4, "err", "error");
+
+    private final int numericValue;
+    private final Set<String> allowedLogLevelVariables;
+
+    LogLevel(int numericValue, String... allowedLogLevelVariables) {
+        this.numericValue = numericValue;
+        this.allowedLogLevelVariables = new HashSet<>(Arrays.asList(allowedLogLevelVariables));
+    }
+
+    /**
+     * Converts the log level into a numeric representation used for comparisons.
+     *
+     * @return The numeric representation of the log level.
+     */
+    public int toNumeric() {
+        return numericValue;
+    }
+
+    /**
+     * Converts the log level into string representations used for comparisons.
+     *
+     * @return The string representations of the log level.
+     */
+    public Set<String> getAllowedLogLevels() {
+        return allowedLogLevelVariables;
+    }
+
+    /**
+     * Convert log level value to LogLevel.
+     *
+     * @param logLevelVal The log level value which needs to convert
+     * @return The LogLevel Enum.
+     */
+    public static LogLevel getLogLevel(String logLevelVal) {
+        return Arrays.stream(LogLevel.values()).filter(logLevel ->
```

Comment:
Any way to convert this into a `Map`? Right now this would require every call to `getLogLevel` (which there will be many) to iterate over the the enum values.

Reply:
Do you have any suggestions? We currently have ERROR(1, "error", "err")
I can have a map of `{1: ERROR} ` Numeric value to LogLevel
The string array value needs to iterate anyway. Let me know if you have good idea.

Reply:
Just have a static Map<Object, LogLevel>.

Reply:
Turn them into Map.